src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -22,12 +23,15 @@
 from ._version import APIVersion
 from .orchestrator import raise_if_no_orchestrator
 
+
 LIST_HOST_SCHEMA = {
     "hostname": (str, "Hostname"),
     "services": ([{
         "type": (str, "type of service"),
         "id": (str, "Service Id"),
     }], "Services related to the host"),
+    "service_instances": (
+        [str], "Service instances related to the host"),
```

Comment:
done this. i changed the version to 1.2. I am still not entirely clear with how the whole versioning works. So if the API version is 1.2, then request comes with 1.1 is also allowed because of backward compatibility? Because, without any frontend changes, I think this is still working. (I am double checking to make sure that I need to do the frontend changes in the request as well)

Reply:
Minor version upgrades (let's say `v1.2`) are for additive changes (a new field is added to the response). That means that a backend that for a given endpoint implements `v1.2`, it also accepts (and return valid responses) for clients requesting that endpoint under `v1.1` and `v1.0`.

However, if an older Dashboard back-end, that only supports `v1.1`, receives a request from a newer client (Dashboard front-end is actually another client) for the `v1.2` version of the endpoint, it should return a failure (`415`).

The infrastructure logic for dealing with this is (almost) already there (@avanthakkar please correct me if I'm wrong), but there's not yet an easy way tag multiple `list()` methods for different versions (e.g.: a different implementation for `list()` method at `v1.1` than the one from `v1.2`)... That means that, yeah, you might be right and the minor version behaviour is already there! 🙈

Reply:
Got it, thanks. based on this I updated the request header version as well from the dashboard.